pylint/__pkginfo__.py

```diff
@@ -27,8 +27,8 @@ def get_numversion_from_version(v: str) -> Tuple:
                 version.append(int(num))
             except ValueError:
                 version.append(0)
-    while len(version) != 3:
-        version.append(0)
+    # pad out version with 0s
+    version += [0] * (3 - len(version))
```

Comment:
Just my opinion, but I would prefer the old way. I find it more intuitive.
Besides, I don't see a good point in changing a method that is only kept for compatibility reasons.

Reply:
Lately I've been exploring the idea of programming using only primitive-recursive operators. Turing-completeness is not often required, and most tasks can be done without it. The `while` operator is awfully powerful, and it's definitely overkill for such a simple task as "pad out this list until it's at least of length 3". Ideally there would be no `while` loops at all except for where they're really needed, like perhaps in inference.

Of course this is an idiosyncratic concern, so I reverted the change.

Reply:
There's this french author, Perec, who wrote a full novel without the letter e in it. It's an awfully powerful letter - the most used one in french in fact - and apparently, often you don't really need it 😄 Anyway I digress, removing part of a language for the beauty of it is interesting as an exercise, but pylint's users won't benefit from it and may even suffer from it because reviewing opinionated changes takes maintainer time that could be used elsewhere (we have 690 issues currently). Please don't refactor working code without a good reason.